src/library-authoring/collections/LibraryCollectionPage.tsx

```diff
@@ -0,0 +1,193 @@
+    openAddContentSidebar,
+  } = useContext(LibraryContext);
+
+  if (!canEditLibrary) {
+    return null;
+  }
+
+  return (
+    <div className="header-actions">
+      <Button
+        className="ml-1"
+        iconBefore={Add}
+        variant="primary rounded-0"
+        onClick={openAddContentSidebar}
+        disabled={!canEditLibrary}
+      >
+        {intl.formatMessage(messages.newContentButton)}
+      </Button>
+    </div>
+  );
+};
+
+const SubHeaderTitle = ({
+  title,
+  canEditLibrary,
+  infoClickHandler,
+}: {
+  title: string;
+  canEditLibrary: boolean;
+  infoClickHandler: () => void;
+}) => {
+  const intl = useIntl();
+
+  return (
+    <Stack direction="vertical">
+      <Stack direction="horizontal" gap={2}>
+        {title}
+        <IconButton
+          src={InfoOutline}
+          iconAs={Icon}
+          alt={intl.formatMessage(messages.collectionInfoButton)}
+          onClick={infoClickHandler}
+          variant="primary"
+        />
+      </Stack>
+      { !canEditLibrary && (
+        <div>
+          <Badge variant="primary" style={{ fontSize: '50%' }}>
+            {intl.formatMessage(messages.readOnlyBadge)}
+          </Badge>
+        </div>
+      )}
+    </Stack>
+  );
+};
+
+const LibraryCollectionPage = () => {
+  const { libraryId, collectionId } = useParams();
+
+  if (!collectionId || !libraryId) {
+    // istanbul ignore next - This shouldn't be possible; it's just here to satisfy the type checker.
+    throw new Error('Rendered without collectionId or libraryId URL parameter');
+  }
+
+  const intl = useIntl();
+
+  const {
+    sidebarBodyComponent,
+    openCollectionInfoSidebar,
+  } = useContext(LibraryContext);
+
+  useEffect(() => {
+    openCollectionInfoSidebar();
+  }, []);
+
+  const { data: libraryData, isLoading: isLibLoading } = useContentLibrary(libraryId);
+  const { data: collectionData, isLoading: isCollectionLoading } = useCollection(libraryId, collectionId);
+
+  if (isLibLoading || isCollectionLoading) {
+    return <Loading />;
+  }
+
+  if (!libraryData || !collectionData) {
+    return <NotFoundAlert />;
+  }
+
+  const breadcrumbs = [
+    {
+      label: libraryData.title,
+      to: `/library/${libraryId}`,
+    },
+    {
+      label: intl.formatMessage(messages.allCollections),
+      to: `/library/${libraryId}/collections`,
+    },
+    // Adding empty breadcrumb to add the last `>` spacer.
+    {
+      label: '',
+      to: '',
+    },
+  ];
+
+  return (
+    <div className="d-flex">
+      <div className="flex-grow-1">
+        <Header
+          number={libraryData.slug}
+          title={libraryData.title}
+          org={libraryData.org}
+          contextId={libraryId}
+          isLibrary
+        />
+        <Container size="xl" className="px-4 mt-4 mb-5 library-authoring-page">
+          <SearchContextProvider
+            extraFilter={[`context_key = "${libraryId}"`, `collections.key = "${collectionId}"`]}
+            fetchCollections={false}
```

Comment:
I don't quite understand why we need this flag if we have to add `useCollection` to fetch the collection anyway?

I would have thought that those `extraFilter`s you're sending to `SearchContextProvider` would ensure that the first result in the collection query (multisearch number 3) is the Collection we want to show here.

Reply:
@pomegranited `useCollection` gets collection data from API while this flag controls fetching of all collections based on search. We only want to fetch blocks under this collection which is why we have update the extraFilter. As I mentioned in the [comment](https://github.com/openedx/frontend-app-authoring/pull/1281#discussion_r1764524711) above, there seems to be some issue with fetching single document from meilisearch.

But I do like the idea of using block_id field which is currently set to `collection.key` but adding `content_key` i.e. library.key into the filter we can get an individual document. If you agree, I can update it and remove the need to calling collections api to fetch single collection data.

Reply:
Oh yes please -- it would be better if we could use meilisearch's data instead of hitting the backend, since we're doing that everywhere else.

Reply:
@pomegranited Done. Had to include `block_id` to filterable attributes in edx-platform and update search manager to override queries.

Reply:
Noted, and thanks for making that change @navinkarkera . We just need to remember to run `reindex_search` on the tagging sandbox when we deploy this change for testing.